### DISCUSSION TITLE
10月24日（火）VRoid Hubメンテナンス実施のお知らせ（2023年10月18日）

### DISCUSSION THREAD

**Discussion by nyamadan** [pixiv/vroid-sdk-developers#113] [Announcements]
VRoid Hubをご利用いただきましてありがとうございます。
下記日程にて、[VRoid Hub](https://hub.vroid.com/) のメンテナンスを実施いたします。

【日時】
2023年10月24日（火）14:30～15:30
※終了時間は予告なく変更させていただく場合がございます。

【影響内容】
メンテナンス中は、以下がご利用になれません。
 * VRoid Hubの閲覧
 * VRoid HubにアップロードしたキャラクターのVRoid Hub連携アプリでの利用

ご不便をおかけしますが、何卒ご理解とご協力をお願いいたします。
引き続きVRoid Hubをよろしくお願いいたします。

**Comment by wcl1207** [comment]
VRoid Hub各status(正常、メンテナンスなど)を取得するためのAPIがありますでしょうか。
目的としては、開発中のアプリ内に、VRoid Hubメンテナンスを検出すると、ユーザーに「VRoid Hubメンテナンス中、しばらく使用できません」というwarning messageを出したいです。

**Reply by nyamadan** [reply]
> VRoid Hub各status(正常、メンテナンスなど)を取得するためのAPIがありますでしょうか。

申し訳ございませんが、こちらのAPIは用意しておりません。

> 目的としては、開発中のアプリ内に、VRoid Hubメンテナンスを検出すると、ユーザーに「VRoid Hubメンテナンス中、しばらく使用できません」というwarning messageを出したいです。

メンテナンス中は[GetAccount](https://developer.vroid.com/sdk/docs/0.3.0/Pixiv.VroidSdk.Api.DefaultApi.html#Pixiv_VroidSdk_Api_DefaultApi_GetAccount_System_Action_Pixiv_VroidSdk_Api_DataModel_Account__System_Action_Pixiv_VroidSdk_Api_DataModel_ApiErrorFormat__)をはじめとしたすべてのVRoid Hub APIは503 ステータスとともに次のようなJSONを返却する予定です。

```json
{"data":{},"error":{"message":"VRoid Hub is under maintenance.","code":"UNDER_MAINTENANCE","details":{}}}
```

VRoid SDKのエラーコールバックより受け取れるはずですので `code` が `UNDER_MAINTENANCE` であれば、メンテナンス中であるという表示が可能かと思います。
